core/src/main/java/org/springframework/security/access/expression/ExpressionUtils.java

```diff
@@ -19,12 +19,42 @@
 import org.springframework.expression.EvaluationContext;
 import org.springframework.expression.EvaluationException;
 import org.springframework.expression.Expression;
+import org.springframework.security.authorization.AuthorizationDecision;
+import org.springframework.security.authorization.ExpressionAuthorizationDecision;
 
 public final class ExpressionUtils {
 
 	private ExpressionUtils() {
 	}
 
+	/**
+	 * Evaluate a SpEL expression and coerce into an {@link AuthorizationDecision}
+	 * @param expr a SpEL expression
+	 * @param ctx an {@link EvaluationContext}
+	 * @return the resulting {@link AuthorizationDecision}
+	 * @since 6.3
+	 */
+	public static AuthorizationDecision evaluate(Expression expr, EvaluationContext ctx) {
+		try {
+			Object result = expr.getValue(ctx);
```

Comment:
I believe we should support `AuthorizationResult` instead of `AuthorizationDecision`. We can even add a new method to `AuthorizationManager` named `authorize` to return an `AuthorizationResult`.

Reply:
For now, I'll make sure that all new APIs use `AuthorizationResult`. We can cast inside of authorization manager instances for now. I think we should add support in `AuthorizationManager` for `AuthorizationResult` in a new ticket.